internal/controller/apiextensions/composite/ready.go

```diff
@@ -77,37 +119,20 @@ func ReadinessChecksFromTemplate(t *v1.ComposedTemplate) []ReadinessCheck {
 	}
 	out := make([]ReadinessCheck, len(t.ReadinessChecks))
 	for i := range t.ReadinessChecks {
-		out[i] = ReadinessCheck{Type: ReadinessCheckType(t.ReadinessChecks[i].Type)}
-		if t.ReadinessChecks[i].FieldPath != "" {
-			out[i].FieldPath = pointer.String(t.ReadinessChecks[i].FieldPath)
-		}
-
-		// NOTE(negz): ComposedTemplate doesn't use pointer values for optional
-		// strings, so today the empty string and 0 are equivalent to "unset".
-		if t.ReadinessChecks[i].MatchString != "" {
-			out[i].MatchString = pointer.String(t.ReadinessChecks[i].MatchString)
-		}
-		if t.ReadinessChecks[i].MatchInteger != 0 {
-			out[i].MatchInteger = pointer.Int64(t.ReadinessChecks[i].MatchInteger)
-		}
+		out[i] = ReadinessCheckFromV1(&t.ReadinessChecks[i])
 	}
 	return out
 }
 
-// ReadinessChecksFromDesired derives readiness checks from the supplied desired
+// ReadinessChecksFromDesiredResource derives readiness checks from the supplied desired
 // resource.
-func ReadinessChecksFromDesired(dr *iov1alpha1.DesiredResource) []ReadinessCheck {
+func ReadinessChecksFromDesiredResource(dr *iov1alpha1.DesiredResource) []ReadinessCheck {
```

Comment:
Does this function need to be renamed?

I have a slight preference for the shorter name - `DesiredResource` already appears in the signature as the type of the argument. I don't feel super strongly, but if you do want to give this a longer name let's be consistent and rename `ReadinessChecksFromTemplate` to `ReadinessChecksFromComposedTemplate` too.

Reply:
> I wouldn't go with ReadinessChecksFromComposedTemplate instead of ReadinessChecksFromTemplate as that would be unnecessary

I would prefer that for consistency, but okay with either.  
Please let me know if you want to make a change or prefer to go as is so we can merge the PR.